ProfileActivity.java

```diff
@@ -0,0 +1,103 @@
+package com.circle.profile.ProfileActivity;
+import com.hootsuite.manager.Manager;
+import com.circle.somelibrary.*;
+import java.util.LinkedList;
+import java.util.List;
+import javax.inject.Inject;
+import butterknife.InjectView;
+import kotlin.Pair;
+
+public final class ProfileListFragment extends Fragment implements SomeInterface {
+    
+    @Inject
+    private ManagerImpl manager;
+    private TextView mCountText;  
+  
+    protected LinkedList<SocialProfile> socialProfileList = new LinkedList<SocialProfile>();
+    Activity mParentActivity;
+    
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        mParentActivity = activity;
+    }
+    
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        socialProfileList.add(getIntent().getExtras().getParcelable("profileList"));
+    }
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState)
+        View view = inflater.inflate(R.id.profile_screen, container, false);
+        return view;
+    }
+    
+    public void onStart() {
+        mCountText = (TextView) findViewById(R.id.text_count);
+        RecyclerView recyclingView = (RecyclerView) findViewById(R.id.recycler_view);
+        
+        mCountText.setText("Count of Profiles " + socialProfiles.size());
+        recyclingView.setAdapter(new SocialProfilesAdapter(socialProfileList));
+        recyclingView.setLayoutManager(new LinearLayoutManager());
+    }
+        
+    /**
+     * Finds and returns the profiles in this fragment that is similar to the similarTo parameter.
+     * @param similarTo A collection of profiles to be similar to
+     */
+    public LinkedList<SocialProfile> similar(ArrayList<SocialProfile> similarTo) {
```

Comment:
Change function name to be more descriptive. Perhaps something like 'findProfilesSimilar'

Reply:
Declare your SocialProfiles at the top of your function, this will help (a tiny bit) with performance, and readability.

Reply:
Reverse the loop, if you're removing things and going forwards, it'll cause you to run into issues.

Or use an iterator.

Reply:
minor nit-pick, change to i and t instead of i and j. Just convention.

Reply:
Change public to private, this probably doesn't need to be public.